--- a/GaudiProfiling/CMakeLists.txt.orig	2019-02-22 15:16:26.309666439 +0100
+++ b/GaudiProfiling/CMakeLists.txt	2019-02-22 15:17:19.720168945 +0100
@@ -4,7 +4,7 @@
 
 if (CMAKE_SYSTEM_NAME MATCHES Linux)
 
-find_package(Boost COMPONENTS python)
+find_package(Boost COMPONENTS python${boost_python_version})
 find_package(PythonLibs)
 find_package(unwind)
 find_package(gperftools)
@@ -29,7 +29,7 @@
                    INCLUDE_DIRS unwind ZLIB)
 
   gaudi_add_python_module(PyCPUFamily src/python/CPUFamily.cpp
-                          LINK_LIBRARIES ${Boost_PYTHON_LIBRARY}
+                          LINK_LIBRARIES ${Boost_PYTHON${boost_python_version}_LIBRARY}
                           INCLUDE_DIRS PythonLibs)
 
   gaudi_add_executable(GaudiGenProfilingHtml src/app/pfm_gen_analysis.cpp
--- a/GaudiExamples/CMakeLists.txt.orig	2019-02-22 15:14:38.950660265 +0100
+++ b/GaudiExamples/CMakeLists.txt	2019-02-22 15:15:27.126111804 +0100
@@ -5,7 +5,7 @@
 find_package(AIDA)
 find_package(HepPDT)
 find_package(ROOT COMPONENTS Tree RIO Hist Net REQUIRED)
-find_package(Boost COMPONENTS python REQUIRED)
+find_package(Boost COMPONENTS python${boost_python_version} REQUIRED)
 find_package(CLHEP)
 find_package(GSL)
 find_package(PythonLibs)
@@ -144,11 +144,9 @@
   file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/HackedStatusCode.h" "${StatusCodeHeaderData}")
 endif()
 
-if(Boost_PYTHON_FOUND)
-  gaudi_add_python_module(PyExample src/PythonModule/*.cpp
-                          LINK_LIBRARIES ${Boost_PYTHON_LIBRARY}
-                          INCLUDE_DIRS Boost PythonLibs)
-endif()
+gaudi_add_python_module(PyExample src/PythonModule/*.cpp
+                        LINK_LIBRARIES ${Boost_PYTHON${boost_python_version}_LIBRARY}
+                        INCLUDE_DIRS Boost PythonLibs)
 
 gaudi_install_python_modules()
 
--- a/GaudiCoreSvc/CMakeLists.txt.orig	2019-02-22 15:15:46.022288899 +0100
+++ b/GaudiCoreSvc/CMakeLists.txt	2019-02-22 15:16:12.652538460 +0100
@@ -2,7 +2,7 @@
 
 gaudi_depends_on_subdirs(GaudiKernel)
 
-find_package(Boost COMPONENTS system filesystem regex thread python)
+find_package(Boost COMPONENTS system filesystem regex thread python${boost_python_version})
 find_package(TBB)
 find_package(PythonLibs)
 find_package(ROOT)
--- a/cmake/GaudiProjectConfig.cmake.orig	2019-02-22 15:44:07.587363211 +0100
+++ b/cmake/GaudiProjectConfig.cmake	2019-02-22 15:44:19.482475766 +0100
@@ -176,6 +176,14 @@
 
 find_package(PythonInterp)
 
+find_package(Boost)
+if((Boost_VERSION GREATER 106700) OR (Boost_VERSION EQUAL 106700))
+  set(boost_python_version "${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+else()
+  set(boost_python_version "")
+endif()
+
+
 #-------------------------------------------------------------------------------
 # gaudi_project(project version
 #               [USE proj1 vers1 [proj2 vers2 ...]]

