--- a/GaudiKernel/GaudiKernel/Property.h.orig	2019-02-22 11:58:33.678888657 +0100
+++ b/GaudiKernel/GaudiKernel/Property.h	2019-02-22 11:59:27.250375692 +0100
@@ -534,13 +534,13 @@
     {
       return value().end();
     }
-    template <class ARG, class T = const ValueType>
-    inline decltype( std::declval<T>()[ARG{}] ) operator[]( const ARG& arg ) const
+    template <class ARG>
+    decltype( auto ) operator[]( const ARG& arg ) const
     {
       return value()[arg];
     }
-    template <class ARG, class T = ValueType>
-    inline decltype( std::declval<T>()[ARG{}] ) operator[]( const ARG& arg )
+    template <class ARG>
+    decltype( auto ) operator[]( const ARG& arg )
     {
       return value()[arg];
     }
