--- a/GaudiKernel/CMakeLists.txt.orig	2019-02-22 16:00:24.011582114 +0100
+++ b/GaudiKernel/CMakeLists.txt	2019-02-22 16:00:44.444774629 +0100
@@ -42,7 +42,7 @@
 
 #---Libraries---------------------------------------------------------------
 gaudi_add_library(GaudiKernel src/Lib/*.cpp
-                  LINK_LIBRARIES ${CMAKE_DL_LIBS} Boost ROOT TBB GaudiPluginService
+                  LINK_LIBRARIES ${CMAKE_DL_LIBS} Boost ROOT TBB GaudiPluginService rt
                   INCLUDE_DIRS Boost ROOT TBB
                   PUBLIC_HEADERS GaudiKernel)
 
--- a/GaudiUtils/src/component/StalledEventMonitor.cpp.orig	2019-02-22 16:19:10.898218494 +0100
+++ b/GaudiUtils/src/component/StalledEventMonitor.cpp	2019-02-22 16:19:27.283373361 +0100
@@ -92,7 +92,7 @@
 
   if ( m_eventTimeout ) {
     // create the watchdog thread
-    m_watchdog.reset( new EventWatchdog( msgSvc(), "EventWatchdog", boost::posix_time::seconds( m_eventTimeout ),
+    m_watchdog.reset( new EventWatchdog( msgSvc(), "EventWatchdog", boost::posix_time::seconds( m_eventTimeout.value() ),
                                          m_stackTrace, m_maxTimeoutCount ) );
 
     // register to the incident service
