--- a/Plugins/TGeoPlugins/CMakeLists.txt
+++ b/Plugins/TGeoPlugins/CMakeLists.txt
@@ -22,7 +22,7 @@ if (BUILD_TGEO_PLUGIN)
     if (${ROOT_VERSION_MINOR} MATCHES "8" )
       target_link_libraries (ACTSTGeoPlugin PUBLIC Geom)
     else (${ROOT_VERSION_MINOR} MATCHES "8" )
-      target_link_libraries (ACTSTGeoPlugin PUBLIC ROOT::Geom)
+      #target_link_libraries (ACTSTGeoPlugin PUBLIC Geom)
     endif (${ROOT_VERSION_MINOR} MATCHES "8" )
